src/test/codewhisperer/util/crossFileContextUtil.test.ts

```diff
@@ -0,0 +1,154 @@
+
+            assert.strictEqual(actual, undefined)
+        })
+    })
+
+    describe('partial support - control group', function () {
+        before(async function () {
+            this.timeout(60000)
+        })
+
+        beforeEach(async function () {
+            tempFolder = (await createTestWorkspaceFolder()).uri.fsPath
+        })
+
+        afterEach(async function () {
+            await closeAllEditors()
+            await globals.context.globalState.update(userGroupKey, undefined)
+        })
+
+        it('should be empty if userGroup is control', async function () {
+            if (!shouldRunTheTest()) {
+                this.skip()
+            }
+
+            await globals.context.globalState.update(userGroupKey, {
+                group: UserGroup.Control,
+                version: extensionVersion,
+            })
+
+            const editor = await openATextEditorWithText('content-1', 'file-1.js', tempFolder, { preview: false })
+            await openATextEditorWithText('content-2', 'file-2.js', tempFolder, { preview: false })
+            await openATextEditorWithText('content-3', 'file-3.js', tempFolder, { preview: false })
+            await openATextEditorWithText('content-4', 'file-4.js', tempFolder, { preview: false })
+
+            await assertTabCount(4)
+
+            const actual = await crossFile.fetchSupplementalContextForSrc(editor, fakeCancellationToken)
+            console.log(`in control test, ${CodeWhispererUserGroupSettings.instance.userGroup}`)
+
+            assert.ok(actual !== undefined && actual.length === 0)
+        })
+    })
+
+    describe('partial support - crossfile group', function () {
+        before(async function () {
+            this.timeout(60000)
+        })
+
+        beforeEach(async function () {
+            tempFolder = (await createTestWorkspaceFolder()).uri.fsPath
+        })
+
+        afterEach(async function () {
+            await closeAllEditors()
+        })
+
+        it('should be non empty if userGroup is crossfile', async function () {
+            if (!shouldRunTheTest()) {
+                this.skip()
+            }
+
+            await globals.context.globalState.update(userGroupKey, {
+                group: UserGroup.CrossFile,
+                version: extensionVersion,
+            })
+
+            const editor = await openATextEditorWithText('content-1', 'file-1.js', tempFolder, { preview: false })
+            await openATextEditorWithText('content-2', 'file-2.js', tempFolder, { preview: false })
+            await openATextEditorWithText('content-3', 'file-3.js', tempFolder, { preview: false })
+            await openATextEditorWithText('content-4', 'file-4.js', tempFolder, { preview: false })
+
+            await assertTabCount(4)
+
+            const actual = await crossFile.fetchSupplementalContextForSrc(editor, fakeCancellationToken)
+            console.log(`in crossfile test, ${CodeWhispererUserGroupSettings.instance.userGroup}`)
```

Comment:
@justinmk3 @JadenSimon, do you know how to avoid the race condition here, I checked the value here is still `Control` which is set up at L68 (same file) https://github.com/aws/aws-toolkit-vscode/pull/3668/files#diff-04181568581b3ad5742017095727f66f2749885f6dab8c555d8d2cb0b565f0ceR68.

Reply:
and also it's affecting other test code path

Reply:
what's your recommendation for this kinda of code `codewhispererUserGroup` which is used by lots of other code path and I need to test all different values of it https://github.com/aws/aws-toolkit-vscode/blob/master/src/codewhisperer/util/userGroupUtil.ts#L14

Reply:
like here I want to test the behavior of `fetchSupplementalContextForSrc` under `Control` and `CrossFile` group

Reply:
@zixlin7